pkg/gateway/operations/base.go

```diff
@@ -40,6 +40,10 @@ const (
 
 type ActionIncr func(action, repository, ref, userID string)
 
+type UploadPathProvider interface {
```

Comment:
Why is the UploadPathProvider interface defined under gateway?

Reply:
I can consolidate the interface definition - but using the above practice remove the dependency of `upload` package in gateway.
But because we already use the implementation directly - I accept and I will consolidate the interface usage.